tests/cluster/tests/23-pubsublocal-slot-migration.tcl

```diff
@@ -0,0 +1,160 @@
+source "../tests/includes/init-tests.tcl"
+
+test "Create a 3 nodes cluster" {
+    cluster_create_with_continuous_slots 3 3
+}
+
+test "Cluster is up" {
+    assert_cluster_state ok
+}
+
+set cluster [redis_cluster 127.0.0.1:[get_instance_attrib redis 0 port]]
+
+test "Migrate a slot, verify client receives unsubscribelocal on primary serving the slot." {
+
+    # Setup the to and from node
+    set channelname mychannel
+    set slot [$cluster cluster keyslot $channelname]
```

Comment:
```suggestion
    set slot [R 0 cluster keyslot $channelname]
```
Which is true in several places, unless you are explicitly trying to setup a new connection.

Reply:
Local variable was useful to not handle the unsubscribe flow in certain cases. Made the cluster variable initialization global and handled the state correctly in methods.